atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/TracingKeyValueService.java

```diff
@@ -363,27 +360,92 @@ public boolean supportsCheckAndSet() {
+                LoggingArgs.tableRefValue(tableRef), values.size())) {
             delegate().putWithTimestamps(tableRef, values);
         }
     }
 
     @Override
     public void truncateTable(TableReference tableRef) {
         //noinspection unused - try-with-resources closes trace
-        try (CloseableTrace trace = startLocalTrace("truncateTable({})", tableRef)) {
+        try (CloseableTrace trace = startLocalTrace("truncateTable({})", LoggingArgs.tableRefValue(tableRef))) {
             delegate().truncateTable(tableRef);
         }
     }
 
     @Override
     public void truncateTables(Set<TableReference> tableRefs) {
         //noinspection unused - try-with-resources closes trace
-        try (CloseableTrace trace = startLocalTrace("truncateTables({})", tableRefs)) {
+        try (CloseableTrace trace = startLocalTrace("truncateTables({})", LoggingArgs.tableRefValues(tableRefs))) {
             delegate().truncateTables(tableRefs);
         }
     }
 
+    private RowColumnRangeIterator wrapRowColumnRangeIteratorWithLogger(RowColumnRangeIterator rowColumnRangeIterator,
+            String tableRef,
+            Iterable<byte[]> rows,
+            int cellBatchHint,
+            long timestamp) {
+
+        return new RowColumnRangeIterator() {
+            @Override
+            public Map.Entry<Cell, Value> next() {
+                //noinspection unused - try-with-resources closes trace
+                try (CloseableTrace trace = startLocalTrace("getRowsColumnRange({}, {} rows, {} hint, ts {}).next()",
+                        tableRef, Iterables.size(rows), cellBatchHint, timestamp)) {
+                    return rowColumnRangeIterator.next();
+                }
+            }
+
+            @Override
+            public boolean hasNext() {
+                //noinspection unused - try-with-resources closes trace
+                try (CloseableTrace trace = startLocalTrace("getRowsColumnRange({}, {} rows, {} hint, ts {}).hasNext()",
+                        tableRef, Iterables.size(rows), cellBatchHint, timestamp)) {
+                    return rowColumnRangeIterator.hasNext();
+                }
+            }
+
+            @Override
+            public void remove() {
+                rowColumnRangeIterator.remove();
+            }
+        };
+    }
+
+    private <T> ClosableIterator<T> wrapClosableIteratorWithLogger(ClosableIterator<T> closableIterator,
+            String methodName, String tableRef, long timestamp) {
+
+        return new ClosableIterator<T>() {
+            @Override
+            public T next() {
+                //noinspection unused - try-with-resources closes trace
+                try (CloseableTrace trace = startLocalTrace("{}({}, ts {}).next()",
+                        methodName, tableRef, timestamp)) {
+                    return closableIterator.next();
+                }
+            }
+
+            @Override
+            public void close() {
+                closableIterator.close();
```

Comment:
we may want to trace the close to know when use of iterator completes (i.e. consumer is finished with paging through everything.

Reply:
we should probably create a span on creation of iterator and finish it on close of iterator